### PR TITLE
Allows to override HTTP/2 on per SNI / context

### PR DESCRIPTION
**Pull request by zwoop**
This means you can now enable H2 per domain, without enabling
HTTP/2 globally in the ports specifications.

### CODE REVIEW

File: iocore/net/P_SNIActionPerformer.h [hunk 1]
```diff
@@ -56,24 +49,33 @@ class ActionItem
...
   SNIAction(Continuation *cont) const override
   {
-    auto ssl_vc     = dynamic_cast<SSLNetVConnection *>(cont);
-    auto accept_obj = ssl_vc ? ssl_vc->accept_object : nullptr;
+    SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(cont);
+    NetAccept *accept_obj     = ssl_vc ? ssl_vc->accept_object : nullptr;
+
     if (accept_obj && accept_obj->snpa && ssl_vc) {
-      if (auto it = snpsMap.find(accept_obj->id); it != snpsMap.end()) {
+      // This is a bit of hack, because ALPN strings are managed via accecpt ports as well as via plugin APIs.
```
Comment (randall) [inline iocore/net/P_SNIActionPerformer.h]: spelling (accepcpt)

File: configs/sni.yaml.default [hunk 2]
```diff
@@ -7,7 +7,7 @@
 # YAML-based Configuration file
 #  Format :
 #  Actions available:
-#    disable_h2 - removes H2 from the protocol list advertised by ATS; parameter required = None, parameters = true or false
+#    http2 - enable or disable HTTP/2 from the protocol list advertised by ATS; parameters = 'on' or 'off' (boolean)
```
Comment (mlibbey) [inline configs/sni.yaml.default]: No longer boolean? What's the default?

File: doc/admin-guide/files/sni.yaml.en.rst [hunk 3]
```diff
@@ -98,11 +98,11 @@ client_key                The file containing the client private key that corres
                           :ts:cv:`proxy.config.ssl.client.private_key.filename` is used.
 
 
-disable_h2                :code:`true` or :code:`false`.
+http2                     :code:`on` or :code:`off`.
```
Comment (mlibbey) [inline doc/admin-guide/files/sni.yaml.en.rst]: Prob should note the default here.

**Comment by bryancall** [comment]
[approve ci autest]

File: configs/sni.yaml.default (hunk 2 above)
Comment (zwoop) [reply configs/sni.yaml.default]: YAML booleans takes On/Off, True/ False, Yes/No etc. :) I think On/Off makes most sense here in the docs :).

File: doc/admin-guide/files/sni.yaml.en.rst (hunk 3 above)
Comment (zwoop) [reply doc/admin-guide/files/sni.yaml.en.rst]: Except, there is no default :) The default varies with whatever the ports specification says. This is really an override. But I can try to add something to explain that.

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by randall** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
This simply doesn't work, the internals don't work as the existing code expects :-/. Don't merge this.

**Pull request closed without merging**